src/plugins/autocompletion_providers/Process.ts

```diff
@@ -0,0 +1,188 @@
+                    .filter(i => !arg.params.includes(i.ruser))
+                    .map(i =>
+                        new Suggestion({value: arg.start + i.ruser, displayValue: i.ruser,
+                            description: "User '" + i.ruser + "' with id '" + i.ruserid + "'",
+                            style: styles.optionValue}));
+    };
+
+const effectiveUserSuggestions = async(context: AutocompletionContext): Promise<Suggestion[]> => {
+        const arg = argInfo(context);
+        const users = await Process.users();
+        return users
+                    .filter(i => !arg.params.includes(i.euser))
+                    .map(i =>
+                        new Suggestion({value: arg.start + i.euser, displayValue: i.euser,
+                            description: "User '" + i.euser + "' with id '" + i.euserid + "'",
+                            style: styles.optionValue}));
+    };
+
+const effectiveGroupSuggestions = async(context: AutocompletionContext): Promise<Suggestion[]> => {
+        const arg = argInfo(context);
+        const groups = await Process.groups();
+        return groups
+                    .filter(i => !arg.params.includes(i.egroup))
+                    .map(i =>
+                        new Suggestion({value: arg.start + i.egroup, displayValue: i.egroup,
+                            description: "Group '" + i.egroup + "' with id '" + i.egroupid + "'",
+                            style: styles.optionValue}));
+    };
+
+const realGroupSuggestions = async(context: AutocompletionContext): Promise<Suggestion[]> => {
+        const arg = argInfo(context);
+        const groups = await Process.groups();
+        return groups
+                    .filter(i => !arg.params.includes(i.rgroup))
+                    .map(i =>
+                        new Suggestion({value: arg.start + i.rgroup, displayValue: i.rgroup,
+                            description: "Group '" + i.rgroup + "' with id '" + i.rgroupid + "'",
+                            style: styles.optionValue}));
+    };
+
+const terminalSuggestions = async(context: AutocompletionContext): Promise<Suggestion[]> => {
+        const arg = argInfo(context);
+        const terminals = await Process.terminals();
+        return terminals
+                .filter(i => !arg.params.includes(i.name))
+                .map(i => new Suggestion({value: arg.start + i.name, displayValue: i.name,
+                            description: "Terminal with name '" + i.name + "'",
+                            style: styles.optionValue}));
+    };
+
+const processSuggestions = async(context: AutocompletionContext): Promise<Suggestion[]> => {
+        const arg = argInfo(context);
+        const processes = await Process.processes();
+        return processes
+                .filter(i => !arg.params.includes(i.pid))
+                .map(i => new Suggestion({value: arg.start + i.pid, displayValue: i.pid,
+                            description: "Process with ID '" + i.pid +
```

Comment:
Please use backticks style strings, like `` `Process with ID ${i.pid}...` `` for these strings. Also the information shown on the left portion of the autocompletion (the actual PID) probably doesn't need to be shown on the right as well. The command is probably the most important thing to show in the details, and the user the second most important thing. See if you can shrink down this text to make sure there is aways space for the command (and hopefully user as well) here:

![image](https://cloud.githubusercontent.com/assets/558242/18610463/e3d49d0a-7cd1-11e6-823a-604f6041ff80.png)

Similarly for the group, etc.